vault/expiration.go

```diff
@@ -562,11 +583,48 @@ func (m *ExpirationManager) Renew(leaseID string, increment time.Duration) (*log
 	return resp, nil
 }
 
+// RestoreTokenCheck verifies that the token is not expired while running in
+// restore mode.  If we are not in restore mode, the lease has already been
+// restored or the lease still has time left, it returns true.
+func (m *ExpirationManager) RestoreTokenCheck(source string, token string) (bool, error) {
+	// Return immediately if we are not in retore mode, expiration manager is
+	// already loaded
+	if !m.inRestoreMode() {
+		return true, nil
+	}
```

Comment:
What if restore mode changes to `0` after this check? I think we need to use a true mutex here.

Reply:
That's a good point.  I'll add the restoreMutex after it determines it is in restore mode.

Reply:
It had it but just a little later.  I moved it up.